transformations.py

```diff
@@ -0,0 +1,114 @@
+        dept_codes = pd.read_csv("dept_codes.csv")
+        self.dataframe = pd.merge(
+            self.dataframe, dept_codes, on="department", how="left"
+        )
+        self.dataframe.drop(columns="Unnamed: 0", inplace=True)
+        new_dept_column = []
+        for result in self.dataframe["dept_code"]:
+            if isinstance(result, float):
+                new_value = str(result)[:4]
+                new_dept_column.append(new_value)
+            else:
+                new_dept_column.append(None)
+
+        self.dataframe["dept_code"] = new_dept_column
+
+    ##Paycom only accepts words and digits within address column so this method removes unacceptable characters
+    def clean_address(self):
+        new_addr_column = []
+        for result in self.dataframe["address"]:
+            new_value = (
+                result.replace(".", "")
+                .replace("#", "")
+                .replace(",", "")
+                .replace("/", "")
+            )
+            new_addr_column.append(new_value)
+        self.dataframe["address"] = new_addr_column
+
+    ##Standardizing homePhone to meet Paycom's format requirement
+    def clean_homePhone(self):
+        new_phone_column = []
+        for result in self.dataframe["homePhone"]:
+            new_value = re.sub("[^0-9]", "", result)
+            new_phone_column.append(new_value)
+        self.dataframe["homePhone"] = new_phone_column
+
+    ##Converting hireDate from Epoch Unix to the date format that Paycom requires
+    def clean_hireDate(self):
+        new_hireDate_column = []
+        for result in self.dataframe["hireDate"]:
+            if len(result) > 1:
+                epoch_integer = int(result[:-3])
+                time_formatted = time.strftime(
+                    "%Y-%m-%d", time.localtime(epoch_integer)
+                )
+                new_hireDate_column.append(time_formatted)
+            else:
+                new_hireDate_column.append(None)
+        self.dataframe["hireDate"] = new_hireDate_column
+
+    """
+    The next two methods simply extraxt the 3-digit code from both pay and work location (ex. takes 218 from "KIPP Bayview Elementary (218)"). Since
+    not all field values are in this format, it first checks if "(" is in the string. I was getting a data type error so I also first checked
+    if the value is a string
+)
+    """
+
+    def parse_pay_location(self):
+        pay_abbr_column = []
+        for result in self.dataframe["assigned_pay_location"]:
+            if isinstance(result, str) and "(" in result:
+                new_value = re.sub("\D", "", result)
+                pay_abbr_column.append(new_value)
+            else:
+                pay_abbr_column.append(None)
+        self.dataframe["pay_location_digit"] = pay_abbr_column
+
+    def parse_work_location(self):
+        location_abbr_column = []
+        for result in self.dataframe["assigned_work_location"]:
+            if isinstance(result, str) and "(" in result:
```

Comment:
Same with type checking. If you are running into instances where the field value isn't a string, I'd recommend just converting to a string rather than checking type. Mixed types in the dataframe will be an issue for importing anyway. Better to explicitly cast.